### ISSUE TITLE
Getting name of self

### ISSUE THREAD

**Issue by JacobHast** [qua-platform/quam#53]
Sometimes I need to get the name of the component I'm working with w.r.t. it's parent, i.e. `self.parent.get_attr_name(self)`. Maybe this could be standard method or property for `QuamComponents` so one doesn't have to remember the above syntax?

**Comment by nulinspiratie** [comment]
Good idea, I was thinking of calling it `QuamComponent.inferred_name`, what do you think?
